frame/transaction-payment/asset-tx-payment/src/lib.rs

```diff
@@ -255,21 +248,19 @@ where
 		info: &DispatchInfoOf<Self::Call>,
 		post_info: &PostDispatchInfoOf<Self::Call>,
 		len: usize,
-		_result: &DispatchResult,
+		result: &DispatchResult,
 	) -> Result<(), TransactionValidityError> {
 		let (tip, who, initial_payment) = pre;
 		match initial_payment {
 			InitialPayment::Native(already_withdrawn) => {
-				let actual_fee = pallet_transaction_payment::Pallet::<T>::compute_actual_fee(
-					len as u32, info, post_info, tip,
-				);
-				<OnChargeTransactionOf<T> as OnChargeTransaction<T>>::correct_and_deposit_fee(
-					&who,
-					info,
-					post_info,
-					actual_fee,
+				pallet_transaction_payment::ChargeTransactionPayment::<T>::do_post_dispatch(
```

Comment:
Should not be necessary if you construct `ChargeTransactionPayment::Pre` (as `(tip, who, already_withdrawn)`), but this is fine as well I would say :shrug:

Reply:
Refactored as suggested, as much simpler indeed.